### PR TITLE
Bump to Iceberg 1.4.0

### PR DESCRIPTION
**Pull request by Fokko**

### CODE REVIEW

File: spark/Dockerfile [hunk 1]
```diff
@@ -59,14 +59,14 @@ RUN mkdir -p ${SPARK_HOME} \
...
 
 # Download iceberg spark runtime
-RUN curl https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12/${ICEBERG_VERSION}/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar -Lo /opt/spark/jars/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar
+RUN curl https://repository.apache.org/content/repositories/orgapacheiceberg-1145/org/apache/iceberg/iceberg-spark-runtime-3.4_2.12/1.4.0/iceberg-spark-runtime-3.4_2.12-1.4.0.jar -Lo /opt/spark/jars/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar
 
-# Download Java AWS SDK
-ENV AWSSDK_VERSION=2.20.18
-RUN curl https://repo1.maven.org/maven2/software/amazon/awssdk/bundle/${AWSSDK_VERSION}/bundle-${AWSSDK_VERSION}.jar -Lo /opt/spark/jars/bundle-${AWSSDK_VERSION}.jar
+# Download Java AWS Bundle
+RUN curl https://repository.apache.org/content/repositories/orgapacheiceberg-1145/org/apache/iceberg/iceberg-aws-bundle/1.4.0/iceberg-aws-bundle-1.4.0.jar -Lo /opt/spark/jars/iceberg-aws-bundle-1.4.0.jar
```
Comment (nastra) [inline spark/Dockerfile:65]: minor: this should probably use `ICEBERG_VERSION` later instead of the hardcoded one

Comment (Fokko) [reply spark/Dockerfile:65]: Just copied it from the email for now :)

Comment (Fokko) [reply spark/Dockerfile]: Updated!

File: spark/Dockerfile [hunk 2]
```diff
@@ -59,14 +59,14 @@ RUN mkdir -p ${SPARK_HOME} \
  && rm -rf spark-${SPARK_VERSION}-bin-hadoop3.tgz
 
 # Download iceberg spark runtime
-RUN curl https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12/${ICEBERG_VERSION}/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar -Lo /opt/spark/jars/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar
+RUN curl https://repository.apache.org/content/repositories/orgapacheiceberg-1146/org/apache/iceberg/iceberg-spark-runtime-3.4_2.12/1.4.0/iceberg-spark-runtime-3.4_2.12-1.4.0.jar -Lo /opt/spark/jars/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar
```
[the remark this answers is not in the document]
Comment (Fokko) [reply spark/Dockerfile]: Thanks @issac-lim That's much nicer with the `-Lo`

File: spark/Dockerfile [hunk 3]
```diff
@@ -77,6 +75,7 @@ RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2
 
 # Add iceberg spark runtime jar to IJava classpath
 ENV IJAVA_CLASSPATH=/opt/spark/jars/*
+ENV JAVA_PATH="/opt/spark/jars/*"
```
Comment (nastra) [inline spark/Dockerfile:78]: is this used anywhere?

File: spark/Dockerfile [hunk 4]
```diff
@@ -59,14 +59,12 @@ RUN mkdir -p ${SPARK_HOME} \
  && rm -rf spark-${SPARK_VERSION}-bin-hadoop3.tgz
 
 # Download iceberg spark runtime
-RUN curl https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12/${ICEBERG_VERSION}/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar -Lo /opt/spark/jars/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar
+RUN curl https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12/${ICEBERG_VERSION}/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar \
+ -Lo /opt/spark/jars/iceberg-spark-runtime-${SPARK_MAJOR_VERSION}_2.12-${ICEBERG_VERSION}.jar
```
Comment (nastra) [inline spark/Dockerfile:63]: nit: I'd prefer to have this on a single line. Same for the AWS Bundle line

File: spark/Dockerfile (hunk 3 above)
Comment (Fokko) [reply spark/Dockerfile]: Not sure, was a while ago. I've removed it

**Review by nastra — APPROVED** [review]
LGTM, thanks @Fokko

**Pull request merged**